test/CommonMUnitTests.hs

```diff
@@ -0,0 +1,58 @@
+module CommonMUnitTests (commonMUnitTests) where
+
+import Prelude ()
+
+import Game.LambdaHack.Core.Prelude
+
+import Test.Tasty
+import Test.Tasty.HUnit
+
+import qualified Data.EnumMap.Strict as EM
+
+
+import           Game.LambdaHack.Client.CommonM
+
+import           Game.LambdaHack.Common.Area
+import           Game.LambdaHack.Common.Kind
+import           Game.LambdaHack.Common.Level
+import           Game.LambdaHack.Common.Perception
+import           Game.LambdaHack.Common.Point
+import           Game.LambdaHack.Common.State
+import           Game.LambdaHack.Content.TileKind
+import           Game.LambdaHack.Common.Time
+
+import qualified Game.LambdaHack.Core.Dice as Dice
+
+import           UnitTestHelpers
+
+testLevel :: Level
+testLevel = Level
+  { lkind = toEnum 0
+  , ldepth = Dice.AbsDepth 1
+  , lfloor = EM.empty
+  , lembed = EM.empty
+  , lbig = EM.empty
+  , lproj = EM.empty
+  , ltile = unknownTileMap (fromJust (toArea (0,0,0,0))) unknownId 10 10
+  , lentry = EM.empty
+  , larea = trivialArea (Point 0 0)
+  , lsmell = EM.empty
+  , lstair = ([],[])
+  , lescape = []
+  , lseen = 0
+  , lexpl = 0
+  , ltime = timeZero
+  , lnight = False
+  }
+
+commonMUnitTests :: TestTree
+commonMUnitTests = testGroup "commonMUnitTests" $ 
+  [ testCase "getPerFid stubCliState returns emptyPerception" $
+    do 
+      result <- executorCli (getPerFid testLevelId) stubCliState
+      fst result @?= emptyPer
+  , testCase "makeLine stubLevel fails" $
```

Comment:
The reason it fails is because the actor stands at the target position (`Point 0 0`) and he can't shoot into himself. It doesn't fail because the level is a stub.